app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEffectHandler.kt

```diff
@@ -1,14 +1,65 @@
 package org.simple.clinic.scheduleappointment
 
 import com.spotify.mobius.rx2.RxMobius
+import dagger.Lazy
 import io.reactivex.ObservableTransformer
+import org.simple.clinic.facility.Facility
+import org.simple.clinic.overdue.AppointmentConfig
+import org.simple.clinic.overdue.PotentialAppointmentDate
+import org.simple.clinic.overdue.TimeToAppointment
+import org.simple.clinic.protocol.Protocol
+import org.simple.clinic.protocol.ProtocolRepository
+import org.simple.clinic.util.Just
+import org.simple.clinic.util.None
+import org.simple.clinic.util.Optional
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.util.plus
+import org.simple.clinic.util.toOptional
+import org.threeten.bp.LocalDate
 import javax.inject.Inject
 
-class ScheduleAppointmentEffectHandler @Inject constructor() {
+class ScheduleAppointmentEffectHandler @Inject constructor(
+    private val currentFacility: Lazy<Facility>,
+    private val protocolRepository: ProtocolRepository,
+    private val appointmentConfig: AppointmentConfig,
+    private val userClock: UserClock
+) {
 
   fun build(): ObservableTransformer<ScheduleAppointmentEffect, ScheduleAppointmentEvent> {
     return RxMobius
         .subtypeEffectHandler<ScheduleAppointmentEffect, ScheduleAppointmentEvent>()
+        .addTransformer(LoadDefaultAppointmentDate::class.java, loadDefaultAppointmentDate())
         .build()
   }
+
+  private fun loadDefaultAppointmentDate(): ObservableTransformer<LoadDefaultAppointmentDate, ScheduleAppointmentEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .map { currentFacility.get() }
+          .map(::currentProtocol)
```

Comment:
These two calls could be merged. Wdyt?
In this chain, we don't care how are we getting the `currentProtocol` so we can move `currentFacility.get()` in this method.

Reply:
They could, I'll do this.

Reply:
Done.